hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/MaintainersEndpoint.java

```diff
@@ -0,0 +1,101 @@
+package pl.allegro.tech.hermes.management.api;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import pl.allegro.tech.hermes.api.ErrorCode;
+import pl.allegro.tech.hermes.api.Maintainer;
+import pl.allegro.tech.hermes.common.exception.HermesException;
+import pl.allegro.tech.hermes.management.domain.maintainer.HintingMaintainerSource;
+import pl.allegro.tech.hermes.management.domain.maintainer.MaintainerSource;
+import pl.allegro.tech.hermes.management.domain.maintainer.MaintainerSourceNotFound;
+import pl.allegro.tech.hermes.management.domain.maintainer.MaintainerSources;
+
+import javax.ws.rs.*;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+@Component
+@Path("/maintainers")
+@Api(value = "/maintainers", description = "Provides maintainers information")
+public class MaintainersEndpoint {
+
+    private MaintainerSources maintainerSources;
+
+    @Autowired
+    public MaintainersEndpoint(MaintainerSources maintainerSources) {
+        this.maintainerSources = maintainerSources;
+    }
+
+    @GET
+    @Path("/sources/{source}")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Lists maintainers from the given source matching the search string", response = List.class, httpMethod = HttpMethod.GET)
+    public List<Maintainer> search(@PathParam("source") String source,
+                                   @QueryParam("search") String searchString) {
+        MaintainerSource maintainerSource = maintainerSources.getByName(source).orElseThrow(() -> new MaintainerSourceNotFound(source));
```

Comment:
Here you can add more specific method to `MaintainerSources` sth like `getHintingMaintainerSource(...)`, then you dont' have to cast in L45